src/ios/CameraRenderController.m

```diff
@@ -58,119 +71,65 @@ - (void)onAppWillResignActive {
+                    [self.mtkView draw];
+                });
+            }
         [self.renderLock unlock];
     }
 }
 
-- (BOOL)shouldAutorotate {
-    return YES;
-}
+- (void)drawInMTKView:(MTKView *)view {
+    if (!self.cameraTexture) return;
+
+    id<MTLCommandBuffer> commandBuffer = [self.commandQueue commandBuffer];
+    MTLRenderPassDescriptor *passDescriptor = view.currentRenderPassDescriptor;
+    if (!passDescriptor) return;
 
--(void) viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator {
-    [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
-    __block UIInterfaceOrientation toInterfaceOrientation;
-    [coordinator animateAlongsideTransition:^(id<UIViewControllerTransitionCoordinatorContext> context) {
-        toInterfaceOrientation = [self.sessionManager getOrientation];
+    id<MTLRenderCommandEncoder> encoder = [commandBuffer renderCommandEncoderWithDescriptor:passDescriptor];
+    [encoder setRenderPipelineState:self.pipelineState];
+    [encoder setFragmentTexture:self.cameraTexture atIndex:0];
 
-    } completion:^(id<UIViewControllerTransitionCoordinatorContext> context) {
-        [self.sessionManager updateOrientation:[self.sessionManager getCurrentOrientation:toInterfaceOrientation]];
-    }];
+    // Draw 2 triangles to make a full-screen quad (4 vertices)
+    [encoder drawPrimitives:MTLPrimitiveTypeTriangleStrip vertexStart:0 vertexCount:4];
+    
+    [encoder endEncoding];
+    [commandBuffer presentDrawable:view.currentDrawable];
+    [commandBuffer commit];
 }
 
--(void) deallocateRenderMemory {
-    if (_renderBuffer) {
-        glDeleteRenderbuffers(1, &_renderBuffer);
-        _renderBuffer = 0;
-    }
-    if (_videoTextureCache) {
-        CVOpenGLESTextureCacheFlush(_videoTextureCache, 0);
-        CFRelease(_videoTextureCache);
-        _videoTextureCache = nil;
-    }
-    if(_lumaTexture) {
-        CVOpenGLESTextureCacheFlush(_lumaTexture, 0);
-        CFRelease(_lumaTexture);
-        _lumaTexture = nil;
+- (void)mtkView:(MTKView *)view drawableSizeWillChange:(CGSize)size {
+    UIInterfaceOrientation orientation = [self.sessionManager getOrientation];
+    [self.sessionManager updateOrientation:[self.sessionManager getCurrentOrientation:orientation]];
+}
+
+- (void)dealloc {
+    if (_textureCache) {
+        CVMetalTextureCacheFlush(_textureCache, 0);
+        CFRelease(_textureCache);
+        _textureCache = nil;
     }
-    [[NSNotificationCenter defaultCenter] removeObserver:self name:UIApplicationDidBecomeActiveNotification object:nil];
-    [[NSNotificationCenter defaultCenter] removeObserver:self name:UIApplicationWillEnterForegroundNotification object:nil];
-    [self.view removeFromSuperview];
-    [EAGLContext setCurrentContext:nil];
-    self.context = nil;
-    self.ciContext = nil;
+    _mtkView.delegate = nil;
+    _cameraTexture = nil;
```

Comment:
We are not using this

Reply:
These variables are currenty used in loadView() and captureOutput()